acoustid/future/fingerprint.py

```
@@ -22,29 +22,43 @@
     compression, packs them into bytes and applies zstd compression.
     """
+    if signed:
+        hashes = to_unsigned(hashes)
     diffs = [hashes[i] ^ (hashes[i - 1] if i > 0 else 0) for i in range(len(hashes))]
-    data = struct.pack(f"<HBB{len(diffs)}i", MAGIC, FORMAT, version, *diffs)
+    data = struct.pack(f"<HBB{len(diffs)}I", MAGIC, FORMAT, version, *diffs)
     return zstd.compress(data, 0)
 
 
-def decompress_fingerprint(data: bytes) -> tuple[list[int], int]:
+def decompress_fingerprint(data: bytes, signed: bool = False) -> tuple[list[int], int]:
```

Ruff (0.8.2)

Boolean-typed positional argument in function definition

(FBT001)